libraries/AP_Param/AP_Param.h

```diff
@@ -210,8 +210,9 @@ class AP_Param
     // a token used for first()/next() state
     typedef struct {
         uint32_t key : 9;
-        uint32_t idx : 5; // offset into array types
+        uint32_t idx : 4; // offset into array types
```

Comment:
I'm not super familiar with the code, but from the outside, it looks like ParamToken.idx used to hold 5bits of data and now only holds 4, is that a safe thing to do?   You've also moved the bit position of group_element down by one,  again, I'm not sure of the risk ..but I imagine a scenario where a ParamToken's 32bits are filled bitwise from a flash that was written bitwise with the old code, and end up bits doing things they shouldn't...?

Reply:
it actually only needs 2 bits. It is used to index into parameter arrays, and the only param arrays we have are AP_Vector3f, which index from 0 to 2.
We had it as 5 bits to make it all neatly 32 bits total.
The ordering is needed so I didn't need to change this table:
https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_OSD/AP_OSD_ParamScreen.cpp#L130

Reply:
also note that ParamToken is in memory only, it isn't a persistent data structure